pkg/kv/kvserver/store_raft.go

```diff
@@ -519,6 +533,41 @@ func (s *Store) processReady(rangeID roachpb.RangeID) {
 	}
 }
 
+type cordonRecord struct {
+	RangeID roachpb.RangeID
+}
+
+func (s *Store) markReplicaForCordoningIfPanic(ctx context.Context, rangeID roachpb.RangeID) {
+	if re := recover(); re != nil {
+		func() {
+			log.Errorf(ctx, "cordoning range %v since panic and/or fatal error during application: %v",
+				rangeID, re)
+
+			// TODO(josh): For the POC, we only allow cordoning a single range at a time.
+			c := cordonRecord{RangeID: rangeID}
+			// TODO(josh): Use proto instead of yaml.
```

Comment:
We should put the range ID into the key as a suffix. In that case I don't think we need a serialized value at all, `nil` should do just fine.